processor/elasticinframetricsprocessor/config.go

```diff
@@ -0,0 +1,9 @@
+package elasticinframetricsprocessor // import "github.com/elastic/opentelemetry-collector-components/processor/elasticinframetricsprocessor"
+
+type Config struct {
+	AddSystemMetrics bool `mapstructure:"add_system_metrics"`
+}
+
+func (c *Config) Validate() error {
```

Comment:
Is this required if it has no actual effect?

Reply:
Good point! It has no effect in the processor module by itself, but it has once you bundle it into a collector. All components within the collector must implement this interface. On start, the collector goes one by one to each component and runs this function to validate the configuration: https://github.com/open-telemetry/opentelemetry-collector/blob/v0.101.0/otelcol/config.go#L45

Reply:
To make this more explicit, what do you think if we add the following type assertion:

```Go
var _ component.Config = (*Config)(nil)
```

Reply:
I had the impression that in some components this Validate() was missing but maybe it was some kind of "inheritance" behind the scenes and that's why that wasn't an issue.

Since it's required, we can keep either way.

Reply:
I will keep the validate code as it is then.